.github/workflows/reusable-porting-guide.yml

```
@@ -0,0 +1,96 @@
+---
+name: Create porting guide PR
+
+"on":
+  workflow_call:
+    inputs:
+      repository:
+        type: string
+        description: The repository to check out.
+        required: true
+      path:
+        type: string
+        description: The path for the repository.
+        required: true
+      ansible-version:
+        type: string
+        description: Ansible release version.
+        required: true
+      ansible-major-version:
+        type: string
+        description: Ansible major version.
+        required: true
+
+env:
+  git-branch: porting-guide-${{ inputs.ansible-version }}
+  porting-guide-rst: porting_guide_${{ inputs.ansible-major-version }}.rst
+  ci-commit-message: >-
+    Add the Ansible community "${{ inputs.ansible-version }}" porting guide
+  pr-body-message: >-
+    ##### SUMMARY
+
+    Add the Ansible "${{ inputs.ansible-major-version }}" porting guide.
+
+    ##### ISSUE TYPE
+
+    - Docs Pull Request
+
+    ##### COMPONENT NAME
+
+    porting_guide_${{ inputs.ansible-major-version }}.rst
+
+jobs:
+  create-pr:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out the repository
+      uses: actions/checkout@v4
+      with:
+        repository: ${{ inputs.repository }}
+        path: ${{ inputs.path }}
+
+    - name: Download the artifact with the RST file
+      uses: actions/download-artifact@v4
+      with:
+        name: ansible-porting-guide
+        path: ${{ inputs.path }}
+
+    - name: Copy the RST file to the docs repo
+      if: ${{ inputs.repository == 'ansible/ansible-documentation' }}
+      run: cp -v ${{ env.porting-guide-rst }} docs/docsite/rst/porting_guides/
+      working-directory: ${{ inputs.path }}
+
+    - name: Copy the RST file to the build-data repo
+      if: ${{ inputs.repository == 'ansible-community/ansible-build-data' }}
+      run: cp -v ${{ env.porting-guide-rst }} ${{ inputs.ansible-major-version }}
+      working-directory: ${{ inputs.path }}
```

These steps are almost identical. Perhaps, it's best to collapse these into a single step, with the target computed based on inputs..

```suggestion
    - name: Copy the RST file to the ${{ inputs.repository }} repo

      run: >-
        cp -v
        '${{ env.porting-guide-rst }}'
        '${{
          inputs.repository == 'ansible/ansible-documentation'
          && 'docs/docsite/rst/porting_guides/'
          || inputs.ansible-major-version
        }}'
      working-directory: ${{ inputs.path }}
```

---

Also removing the whole creation of  `ansible-build-data` PR code, as it is a duplication, more explanation here in https://github.com/ansible/ansible-documentation/pull/121#discussion_r1628329809